backup/restore.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+this_script="/opt/backup/restore.sh"
+
+set -euo pipefail
+
+backup_file="${1:-}"
+
+if [ ! -f "${backup_file}" ] ; then
+    echo "ERROR: missing backup tarball file."
+    echo "Usage: $0 <backup-tarball>"
+    echo "                  <backup-tarball> - backup created with 'backup.sh'"
+    echo
+    exit 1
+fi
+
+# Only sith lords and backup files deal in absolutes.
+backup_file="$(realpath "${backup_file}")"
+
+if    [[ "${backup_file}" == /opt/mastodon/* ]] \
+   || [[ "${backup_file}" == /opt/backup/* ]] \
+   || [[ "${backup_file}" == /opt/caddy/* ]] ; then
+   echo "ERROR: backup file '${backup_file}' resides in a path that wiĺl be WIPED!"
+   echo "       Move outside /opt/mastodon, /opt/backup, or /opt/caddy and try again."
+   echo
+   exit 1
+fi
+
+
+reinit_script="/opt/backup/re_initialise.sh"
+
+if [ "$(realpath "$0")" = "$this_script" ] ; then
+    # The wiping below will also remove this very script, so we create a tempfile to run from.
+    tmp_restore_script="$(mktemp)"
+    chmod 700 "${tmp_restore_script}"
+    cat "${this_script}" >> "${tmp_restore_script}"
+    echo "Moving to tempfile '${tmp_restore_script}'"
+    exec "${tmp_restore_script}" "${@}"
+fi
+
+trap "rm -f '${0}'" EXIT
+
+if [ ! -f "${reinit_script}" ] ; then
+    echo "INTERNAL ERROR: re-init script not present."
+    exit 2
+fi
+
+tmp_reinit_script="$(mktemp)"
+chmod 700 "${tmp_reinit_script}"
+cat "${reinit_script}" >> "${tmp_reinit_script}"
+
+trap "rm -f '${0}' ${tmp_reinit_script}'" EXIT
```

Comment:
Unbalanced single quotes:

```suggestion
trap "rm -f '${0}' '${tmp_reinit_script}'" EXIT
```